### ISSUE TITLE
DISCUSSION : JS modularization

### ISSUE THREAD

**Issue by issuemover631** [airsonic/airsonic#475]
<a href="https://github.com/biconou"><img src="https://avatars0.githubusercontent.com/u/5565107?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [biconou](https://github.com/biconou)**
_Monday Jul 03, 2017 at 20:27 GMT_
_Originally opened as https://github.com/Libresonic/libresonic/issues/475_

----

Based on the PR #473 I proposed to improve JS code architecture, a discussion started between @muff1nman @fxthomas and I about the best JS modularization solution for the future of Libresonic. 
I'm working on that topic for a few days and I finally catched an interesting Idea. 

Main modularization solutions are 
- commonJS
- AMD (requireJS)
- ES6 modules

Following this thread, https://www.reddit.com/r/javascript/comments/3vqgai/eli5_commonjs_vs_amd_vs_es6/ we can conclude that the ahead solution is definitely ES6. 

_"AMD is dead [...]. CommonJS is dead [...]
ES6 imports aren't currently natively supported in any JS engine I know of. They can however be translated to Node-style "CommonJS" and there are bundlers that support them directly."_

I am studying at the moment a very powerful solution called webpack 2.
Indeed, webpack uses the ES6 module import/export syntax to bundle JS modules in a elegant and secure way. In addition, webpack's capabilities are extremely wide as it aims to bundle all kind of web resources. 

I'll work on a proposal using webpack for Libresonic and submit a new PR when possible. 

What are your thoughts about that ? 

To learn more on webpack please refer to this excellent tutorial : https://www.youtube.com/watch?v=GU-2T7k9NfI&index=15&list=PLmMqTMZttQPZNJ7mFyMQ6YN0Xw94oFDdr.

**Comment by jooola** [comment]
@biconou Is this obsolete now that we have a new UI ?
Closing this !
